README.md

```diff
@@ -1,372 +1,332 @@
 
 ```js
-dbPromise.then(db => {
-  const tx = db.transaction('objs', 'readwrite');
-  tx.objectStore('objs').put({
-    id: 123456,
-    data: {foo: "bar"}
-  });
-  return tx.complete;
-});
+const wrapped = wrap(unwrapped);
 ```
 
-### Getting all
+Use this to convert a plain IDB object to one enhanced by this library. This is useful if some third party code gives you an `IDBDatabase` object and you want it to have the features of this library.
 
-```js
-dbPromise.then(db => {
-  return db.transaction('objs')
-    .objectStore('objs').getAll();
-}).then(allObjs => console.log(allObjs));
-```
+This doesn't work with `IDBCursor`, [due to missing primitives](https://github.com/w3c/IndexedDB/issues/255). Also, if you wrap an `IDBTransaction`, `tx.store` and `tx.objectStoreNames` will not work in Edge. To avoid these issues, wrap the `IDBDatabase` object, and use the wrapped object to create a new transaction.
 
-### Getting by ID
+## General enhancements
+
+Once you've opened the database the API is the same as IndexedDB, except for a few changes to make things easier.
+
+Any method that usually returns an `IDBRequest` object will now return a promise for the result.
 
 ```js
-dbPromise.then(db => {
-  return db.transaction('objs')
-    .objectStore('objs').get(123456);
-}).then(obj => console.log(obj));
+const store = db.transaction(storeName).objectStore(storeName);
+const value = await store.get(key);
 ```
 
-# Limitations
+## `IDBDatabase` enhancements
 
-## Transaction lifetime
+### Shortcuts to get/set from an object store
 
-An IDB transaction will auto-close if it doesn't have anything to do once microtasks have been processed. As a result, this works fine:
+It's common to create a transaction for a single action, so helper methods are included for this:
 
 ```js
-dbPromise.then(async db => {
-  const tx = db.transaction('keyval', 'readwrite');
-  const store = tx.objectStore('keyval');
-  const val = await store.get('counter') || 0;
-  store.put(val + 1, 'counter');
-  return tx.complete;
-});
+// Get a value from a store:
+const value = await db.get(storeName, key);
+// Set a value in a store:
+await db.put(storeName, value, key);
 ```
 
-But this doesn't:
+The shortcuts are: `get`, `getKey`, `getAll`, `getAllKeys`, `count`, `put`, `add`, `delete`, and `clear`. Each method takes a `storeName` argument, the name of the object store, and the rest of the arguments are the same as the equivalent `IDBObjectStore` method.
 
-```js
-dbPromise.then(async db => {
-  const tx = db.transaction('keyval', 'readwrite');
-  const store = tx.objectStore('keyval');
-  const val = await store.get('counter') || 0;
-  // The transaction will auto-close while the fetch is in-progress
-  const newVal = await fetch('/increment?val=' + val)
-  store.put(newVal, 'counter');
-  return tx.complete;
-});
-```
+These methods depend on the same methods on `IDBObjectStore`, therefore `getKey`, `getAll`, and `getAllKeys` are missing in Edge as it lacks support.
```

Comment:
Didn't v3 polyfill these to some degree? Do you think it's worth adding an additional export folks can use if they need Edge support?

Reply:
Yeah, it polyfilled `getAll`. I'm going to hold off on adding too much stuff just for Edge, especially since they're switching to Chromium.